src/main/java/org/springframework/data/r2dbc/repository/support/R2dbcRepositoryFactory.java

```diff
@@ -77,7 +78,12 @@ public R2dbcRepositoryFactory(DatabaseClient databaseClient,
 		this.databaseClient = databaseClient;
 		this.mappingContext = mappingContext;
 		this.dataAccessStrategy = dataAccessStrategy;
-		this.converter = new MappingR2dbcConverter(new BasicRelationalConverter(mappingContext));
+
+		if (dataAccessStrategy instanceof DefaultReactiveDataAccessStrategy) {
```

Comment:
I don't like the check against a specific implementation.
I'd rather have an interface (e.g. `WithConverter`) which other implementations can use to denote that they use a `R2dbcConverter` inside which we can extract.